libass/ass.c

```diff
@@ -660,38 +657,37 @@ static inline void check_duplicate_info_line(const ASS_Track *const track,
         track->parser_priv->header_flags |= si;
 }
 
-static int process_info_line(ASS_Track *track, char *str)
+static int process_info_line(ASS_Track *track, ASS_StringView str)
 {
-    if (!strncmp(str, "PlayResX:", 9)) {
+    if (GATHER_PREFIX(str, "PlayResX:")) {
         check_duplicate_info_line(track, SINFO_PLAYRESX, "PlayResX");
-        track->PlayResX = atoi(str + 9);
-    } else if (!strncmp(str, "PlayResY:", 9)) {
+        track->PlayResX = parse_int_header(str);
+    } else if (GATHER_PREFIX(str, "PlayResY:")) {
         check_duplicate_info_line(track, SINFO_PLAYRESY, "PlayResY");
-        track->PlayResY = atoi(str + 9);
-    } else if (!strncmp(str, "Timer:", 6)) {
+        track->PlayResY = parse_int_header(str);
+    } else if (GATHER_PREFIX(str, "Timer:")) {
         check_duplicate_info_line(track, SINFO_TIMER, "Timer");
-        track->Timer = ass_atof(str + 6);
-    } else if (!strncmp(str, "WrapStyle:", 10)) {
+        // Not parsed, as this doesn't actually do anything
+    } else if (GATHER_PREFIX(str, "WrapStyle:")) {
         check_duplicate_info_line(track, SINFO_WRAPSTYLE, "WrapStyle");
-        track->WrapStyle = atoi(str + 10);
-    } else if (!strncmp(str, "ScaledBorderAndShadow:", 22)) {
+        track->WrapStyle = parse_int_header(str);
+    } else if (GATHER_PREFIX(str, "ScaledBorderAndShadow:")) {
         check_duplicate_info_line(track, SINFO_SCALEDBORDER,
                                     "ScaledBorderAndShadow");
-        track->ScaledBorderAndShadow = parse_bool(str + 22);
-    } else if (!strncmp(str, "Kerning:", 8)) {
+        track->ScaledBorderAndShadow = parse_bool(str);
+    } else if (GATHER_PREFIX(str, "Kerning:")) {
         check_duplicate_info_line(track, SINFO_KERNING, "Kerning");
-        track->Kerning = parse_bool(str + 8);
-    } else if (!strncmp(str, "YCbCr Matrix:", 13)) {
+        track->Kerning = parse_bool(str);
+    } else if (GATHER_PREFIX(str, "YCbCr Matrix:")) {
         check_duplicate_info_line(track, SINFO_COLOURMATRIX, "YCbCr Matrix");
-        track->YCbCrMatrix = parse_ycbcr_matrix(str + 13);
-    } else if (!strncmp(str, "Language:", 9)) {
+        track->YCbCrMatrix = parse_ycbcr_matrix(str);
+    } else if (GATHER_PREFIX(str, "Language:")) {
         check_duplicate_info_line(track, SINFO_LANGUAGE, "Language");
-        char *p = str + 9;
-        while (*p && ass_isspace(*p)) p++;
+        vskip_spaces(&str);
         free(track->Language);
-        track->Language = strndup(p, 2);
-    } else if (!strncmp(str, "; Script generated by ", 22)) {
-        if (!strncmp(str + 22,"FFmpeg/Lavc", 11))
+        track->Language = strndup(str.str, FFMIN(str.len, 2));
+    } else if (GATHER_PREFIX(str, "; Script generated by ")) {
```

Comment:
There are many places where case-sensitive `gather_prefix()` would be useful, I think it should be added to `ass_string.h`. Actually it can fully replace `ass_sv_startswith()`.